src/Common/src/Interop/Unix/System.Native/Interop.readdir.cs

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Runtime.InteropServices;
+using Microsoft.Win32.SafeHandles;
+
+internal static partial class Interop
+{
+    internal static partial class Sys
+    {
+        private static readonly int s_direntSize = GetDirentSize();
+
+        internal enum NodeType : short
+        {
+            DT_UNKNOWN  =  0,
+            DT_FIFO     =  1,
+            DT_CHR      =  2,
+            DT_DIR      =  4,
+            DT_BLK      =  6,
+            DT_REG      =  8,
+            DT_LNK      = 10,
+            DT_SOCK     = 12,
+            DT_WHT      = 14
+        }
+
+        [StructLayout(LayoutKind.Sequential)]
+        private unsafe struct InternalDirectoryEntry
+        {
+            internal NodeType   InodeType;
```

Comment:
Ah sorry, I didn't mean conflict in a compilation way. I try to avoid naming things similar to predefined types (like Type) for logical consistency. I would prefer to keep this as InodeType to be super clear it isn't the same kind of Type, but I can definitely change it if we feel it is better off as Type